tfjs-backend-webgpu/src/kernels/FromPixelsExternalImage.ts

```diff
@@ -16,26 +16,36 @@
  */
 
 import {FromPixelsAttrs, TensorInfo, util} from '@tensorflow/tfjs-core';
+
 import {WebGPUBackend} from '../backend_webgpu';
+
+import {FromPixelsImportProgram} from './FromPixels_utils/from_pixels_import_webgpu';
 import {FromPixelsProgram} from './FromPixels_utils/from_pixels_webgpu';
 import * as webgpu_program from './webgpu_program';
 
+type ExternalImage = HTMLCanvasElement|ImageBitmap|OffscreenCanvas;
+
 export function fromPixelsExternalImage(args: {
-  externalImage: HTMLCanvasElement|ImageBitmap|OffscreenCanvas,
+  externalImage: ExternalImage|HTMLVideoElement,
   backend: WebGPUBackend,
-  attrs: FromPixelsAttrs
+  attrs: FromPixelsAttrs,
+  useImport: boolean
 }): TensorInfo {
-  const {externalImage, backend, attrs} = args;
+  const {externalImage, backend, attrs, useImport} = args;
   const {numChannels} = attrs;
 
   const outShape = [externalImage.height, externalImage.width, numChannels];
   const size = util.sizeFromShape(outShape);
   const strides = util.computeStrides(outShape);
   const uniformData = [size, numChannels, ...strides];
-
   const output = backend.makeTensorInfo(outShape, 'int32');
+
   if (!backend.fromPixelProgram) {
```

Comment:
Will it be a problem if we switch our input source type? For example, in one page, we first use HTMLVideoElement as the input to run a model. Then we use HTMLCanvasElement as the input to run the same model or another model. In the second time, `backend.fromPixelProgram` will be a non-null value and won't call `new FromPixelsProgram`. It seems that we should keep `FromPixelsImportProgram` and `FromPixelsProgram` instance simultaneously rather than one of them.

Reply:
Good point and it indeed happens. Let me think a way to refactor it. We relies on they have same API to reduce duplications.

Reply:
Resolved!